client-course-schedulizer/src/components/Tabs/FacultyLoads/FacultyLoads.tsx

```diff
@@ -0,0 +1,173 @@
+import {
+  Paper,
+  Table,
+  TableBody,
+  TableCell,
+  TableContainer,
+  TableHead,
+  TableRow,
+} from "@material-ui/core";
+import React, { useContext, useMemo } from "react";
+import { Column, useTable } from "react-table";
+import { Schedule, Term } from "../../../utilities/interfaces/dataInterfaces";
+import { ScheduleContext } from "../../../utilities/services/context";
+
+interface TableData {
+  faculty: string;
+  fallCourseSections?: string;
+  fallHours?: number;
+  loadNotes?: string;
+  otherDuties?: string;
+  otherHours?: number;
+  springCourseSections?: string;
+  springHours?: number;
+  summerCourseSections?: string;
+  summerHours?: number;
+  totalHours?: number;
+}
+
+const createTable = (schedule: Schedule): TableData[] => {
+  const newTableData: TableData[] = [];
+  schedule.courses.forEach((course) => {
+    course.sections.forEach((section) => {
+      section.instructors.forEach((instructor) => {
+        const instructorFullName = `${instructor.firstName} ${instructor.lastName}`;
+        const sectionName = `${course.prefixes[0]}-${course.number}-${section.letter}`;
+        const newDataRow: TableData = {
+          faculty: instructorFullName,
+        };
+        const [oldDataRow] = newTableData.filter((data) => {
```

Comment:
A little confused by this. Is this saying that the first element of newTableData.filter() is the oldDataRow as in the previous row of data? Am I understanding that incorrectly?

Reply:
No it's a data row that's been updated previously because a section has already been added for that professor. I'll try to think of a better name.